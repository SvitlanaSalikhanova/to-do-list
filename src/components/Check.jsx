/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import CheckCss from '../styles/Check.module.scss';

export default function Check({ item, onChange }) {
    return (
        <div className={CheckCss.container}>
            <div className={CheckCss.round}>
                <input
                    type="checkbox"
                    checked={item.isDone}
                    onChange={onChange}
                    id={`checkbox${item.id}`}
                />
                <label htmlFor={`checkbox${item.id}`} />
            </div>
        </div>
    );
}
