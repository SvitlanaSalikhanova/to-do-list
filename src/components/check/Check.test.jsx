import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Check from './Check';

const UncompletedItem = {
    id: 1,
    text: 'text',
    isDone: false,
};
const CompletedItem = {
    id: 1,
    text: 'text',
    isDone: false,
};

const onChangeMock = jest.fn();

describe('Check component tests', () => {
    it('Unchecked Matches Snapshot', () => {
        const domTree = renderer.create(
            <Check item={UncompletedItem} onChange={onChangeMock} />,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('Checked Matches Snapshot', () => {
        const domTree = renderer.create(
            <Check item={CompletedItem} onChange={onChangeMock} />,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('when clicked calls onChange', () => {
        render(
            <Check item={UncompletedItem} onChange={onChangeMock} />,
        );
        const check = screen.getByRole('checkbox');
        fireEvent.click(check);
        expect(onChangeMock).toBeCalledTimes(1);
    });
});
