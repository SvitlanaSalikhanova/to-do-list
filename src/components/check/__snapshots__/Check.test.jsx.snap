// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check component tests Checked Matches Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="round"
  >
    <input
      checked={false}
      id="checkbox1"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <label
      htmlFor="checkbox1"
    />
  </div>
</div>
`;

exports[`Check component tests Unchecked Matches Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="round"
  >
    <input
      checked={false}
      id="checkbox1"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <label
      htmlFor="checkbox1"
    />
  </div>
</div>
`;
